### PR TITLE
refactor: position owner check at collection time and reduce boilerplate in fees

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #4971

## What is the purpose of the change

I should not be able to claim fees or incentives if I'm not a position owner. This change adds a relevant changes and tests for it.

Additionally, it reduces boilerplate by introducing an internal `getClaimableFees` function.

## Brief Changelog

- reduce boilerplate
- add owner check
- add tests
- clean up tests

## Testing and Verifying

This change added tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/fees.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/fees.go:231]: I am personally highly against removing comments like these. It allows me to quickly skim the intent of each line, then re-read and check if the function is doing what each line intended.

Will not block on this as its clearly a preference thing, but I will forever write code in this way.

File: x/concentrated-liquidity/fees.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/fees.go:255]: I do not like the way this was separated. Any get function should not have the ability to change state, even if its private. In the future, if someone sees this "getClaimableFees" private methods and attempts to use it in a new function, it would actually claim the rewards. We should make it so that there is no way for the getter to modify state, even if misused.

File: x/concentrated-liquidity/position.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/position.go:92]:
```suggestion
// isPositionOwner returns true if the given positionId is owned by the given sender inside the given pool.
```

**Review by czarcas7ic — COMMENTED** [review]
Good catch on this! Would like a discussion on the getter function not using cache ctx before approving.

File: x/concentrated-liquidity/fees.go
Comment (p0mvn) [reply x/concentrated-liquidity/fees.go:255]: Modifying state is intentional since it is used by collectFees as well. I can change the name to reflect that this is not a getter

File: x/concentrated-liquidity/fees.go
Comment (p0mvn) [reply x/concentrated-liquidity/fees.go:231]: Note that I did not remove any comments that still apply. I just did not add any comments for the new logic since it's trivial

I reduced code duplication by introducing an unexported getClaimableFees. So the old comments do not apply here anymore.

Prior to this change, GetClaimableFees and CollectFees were essentially the same with the exception of cache context

File: x/concentrated-liquidity/fees.go
Comment (czarcas7ic) [reply x/concentrated-liquidity/fees.go:255]: If the name changes then I am good with that!

**Review by czarcas7ic — APPROVED** [review]
Approving since agreement on changing the getter name to something else

File: x/concentrated-liquidity/msg_server_test.go
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/msg_server_test.go:252]: Shouldn't this error message contain the specific position ID and sender (using whichever of the two positions it errors on) as per the error type?

File: x/concentrated-liquidity/msg_server_test.go
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/msg_server_test.go:164]: Out of curiosity, what was the reasoning behind moving this line above the pool/position setup?

File: x/concentrated-liquidity/fees.go
Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/fees.go:255]: +1 on this change

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM once the function name change is made

File: x/concentrated-liquidity/msg_server_test.go
Comment (p0mvn) [reply x/concentrated-liquidity/msg_server_test.go:240]: I opted in for `ErrorAs` check which only checks the error type:
https://github.com/osmosis-labs/osmosis/blob/6270bc674f24f9352a5693efe3518b7d6f4542bc/x/concentrated-liquidity/msg_server_test.go#L196

The reason is that we run `SetupTest()` for every test case. As a result, `s.TestAccs[0]` gets reset on every iteration.

While checking that there are specific arguments formatted to the error is stricter than just the error type, I went with the faster approach to set it up since the marginal benefit of validating the formatting is insignificant IMO.

File: x/concentrated-liquidity/fees.go
Comment (p0mvn) [reply x/concentrated-liquidity/fees.go]: Changed to `prepareClaimableFees` and added godoc

File: x/concentrated-liquidity/msg_server_test.go
Comment (p0mvn) [reply x/concentrated-liquidity/msg_server_test.go:152]: No reason. This can be moved back if desired.

I think I might have suspected that ordering was causing some of the initial issues I was debugging and didn't eventually revert it.

**Comment by p0mvn** [comment]
Merging since all comments are addressed

**Pull request merged**
